### PR TITLE
Ready to be reviewed.

### PR DESCRIPTION
**Pull request by fedirsmilianets**

### CODE REVIEW

File: config/rest-api.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+router.assignRoute(
+  'POST',
+  '/node/deploy/fetch',
+  jwtAuth(bodyUnpacker(localNode.fetchDeploy))
+);
+router.assignRoute(
+  'DELETE',
+  '/node/delete/:id',
+  jwtAuth(bodyUnpacker(localNode.deleteNode))
+);
```
Comment (tshemsedinov) [inline config/rest-api.js]:
```
{
  '/node/deploy/fetch': {
    POST: jwtAuth(bodyUnpacker(localNode.fetchDeploy))
  },
  '/node/delete/:id': {
    DELETE: jwtAuth(bodyUnpacker(localNode.deleteNode))
  }
}
```

File: lib/error-res.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+'use strict';
+
```
Comment (tshemsedinov) [inline lib/error-res.js:2]: To a single handler

File: lib/jstp.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+   */
+  isInitialized(jstpLogin, deployId) {
+    return this._callProcedure(jstpLogin, deployId, 'isInitialized');
+  }
+}
+
+const instance = new JSTPServer();
+Object.freeze(instance);
+
+module.exports = instance;
```
Comment (tshemsedinov) [inline lib/jstp.js:157]: Need no singleton here

File: routes/user.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+  const u = new User(newUser);
+  User.count({}, (err, c) => {
+    if (err) {
+      errors.endBadRequest(res);
+    }
+    if (c === 0) {
+      u.isAdmin = true;
+    } else {
+      u.isAdmin = false;
+    }
```
Comment (tshemsedinov) [inline routes/user.js]: `u.isAdmin = c === 0;`

**Review by tshemsedinov — COMMENTED** [review]
Try to clear code and combine similar parts to a higher order abstractions

**Comment by revan730** [comment]
@tshemsedinov fixes are ready

File: config/rest-api.js [hunk 5]
```diff
@@ -36,101 +27,69 @@ function noMethodHandler(path, method, req, res) {
...
-router.assignRoute(
-  'DELETE',
-  '/node/delete/:id',
-  jwtAuth(bodyUnpacker(localNode.deleteNode))
-);
 
-router.assignRoute('POST', '/webhooks', bodyUnpacker(localNode.webhooks));
+for (const route in routes) {
+  for (const method in routes[route]) {
+    router.assignRoute(method, route, routes[route][method]);
```
Comment (tshemsedinov) [inline config/rest-api.js]: After you converted routing to declarative structure calls `router.assignRoute` will just move this routs to another declarative structure inside router, so maybe you can declare routs in the same structure or rewrite router to use your original declarative routing to remove this calls `router.assignRoute`

File: lib/error-res.js [hunk 6]
```diff
@@ -1,48 +1,80 @@
...
 * Ends request with a 401
 * @param {Response} res
 */
-const endUnauthorised = res => {
-  res.statusCode = 401;
-  res.statusMessage = 'Unauthorised';
-  res.write(JSON.stringify({ error: '401 Unauthorised' }));
-  res.end();
-};
+const endUnauthorised = handlerGenerator(
```
Comment (tshemsedinov) [inline lib/error-res.js]: Even in single line `const endUnauthorised = handlerGenerator(401, 'Unauthorised');`

File: lib/error-res.js [hunk 7]
```diff
@@ -1,48 +1,80 @@
...
+// (res, err) => {
+//   console.error(err);
+//   res.statusCode = 500;
+//   res.statusMessage = 'Server Error';
+//   res.write('500 Server Error');
+//   res.end();
+// };
 
 module.exports = {
   endUnauthorised,
```
Comment (tshemsedinov) [inline lib/error-res.js]: You may move those single line declaration here :)

File: routes/local-node.js [hunk 8]
```diff
@@ -181,6 +182,29 @@ function getDeploy({ req, res, body, match, user }) {
   });
 }
 
+function deployActionGenerator(actionCallback) {
+  return ({ req, res, body, match, user }) => {
+    const { nodeId, deployId } = JSON.parse(body);
+    LocalNode.findById(nodeId, (err, node) => {
+      if (err) {
```
Comment (tshemsedinov) [inline routes/local-node.js]: Single line will be better

**Review by tshemsedinov — CHANGES_REQUESTED** [review]

File: lib/error-res.js (hunk 6 above)
Comment (fedirsmilianets) [reply lib/error-res.js]: Single line tends to exceed 80 chars

File: routes/local-node.js (hunk 8 above)
Comment (fedirsmilianets) [reply routes/local-node.js]: Sorry, I don't get where exactly.

File: lib/error-res.js (hunk 7 above)
Comment (fedirsmilianets) [reply lib/error-res.js]: Unfortunately, jsdoc is not working inside of object declaration.

**Review by tshemsedinov — APPROVED** [review]
:white_flower: ☭ :white_flower: ☭ :white_flower: ☭ :white_flower: ☭ :white_flower: ☭ :white_flower: ☭ :white_flower: ☭ :white_flower: ☭ :white_flower: ☭ :white_flower: ☭

**Pull request merged**